src/components/KCollapse/KCollapse.vue

```diff
@@ -0,0 +1,243 @@
+<template>
+  <div class="k-collapse w-100">
+    <div
+      :class="{
+        'd-flex': trailingTrigger,
+        'd-block': !trailingTrigger
+      }"
+      class="k-collapse-heading w-100 mb-3"
+    >
+      <div
+        v-if="title"
+        :class="{
+          'mr-auto': trailingTrigger,
+          'mb-2': !trailingTrigger
+        }"
+        class="k-collapse-title"
+        data-testid="k-collapse-title"
+      >
+        {{ title }}
+      </div>
+      <div
+        :class="{
+          'ml-auto': trailingTrigger
+        }"
+        class="k-collapse-trigger"
+      >
+        <slot
+          name="trigger"
+          :is-collapsed="collapsedState"
+          :toggle="toggleDisplay"
+        >
+          <a
+            class="k-collapse-trigger-content"
+            data-testid="k-collapse-trigger-content"
+            @click.prevent.stop="toggleDisplay()"
+          >
+            <slot name="trigger-content">
+              <span
+                v-if="triggerLabel"
+                class="k-collapse-trigger-label"
+                data-testid="k-collapse-trigger-label"
+              >
+                <KIcon
+                  :icon="collapsedState ? 'chevronRight' : 'chevronDown'"
+                  size="14"
+                  class="k-collapse-trigger-chevron mr-1"
+                />
+                {{ triggerLabel }}
+              </span>
+              <KIcon
+                v-else
+                :icon="collapsedState ? 'chevronRight' : 'chevronDown'"
+                class="k-collapse-trigger-icon k-collapse-trigger-chevron"
+                data-testid="k-collapse-trigger-icon"
+              />
+            </slot>
+          </a>
+        </slot>
+      </div>
+    </div>
+    <div
+      v-if="hasVisibleContent"
+      class="k-collapse-visible-content w-100 mb-4"
+      data-testid="k-collapse-visible-content"
+    >
+      <slot name="visible-content" />
+    </div>
+    <Transition name="collapse">
```

Comment:
I think I agree with you and don't love the transition we're using at the moment.

Reply:
Removed